Modify Places table and create response to frontend

ai-n0808 opened the pull request:
- Add google place id to table
- Modify newPlace
- Fix TypeScript error

Now it's suppose to do these:
- Google place id is in Places table
- It's checking whether google place id is exist in places table or not
- If not, it's create new place in places table.

ai-n0808 committed:
feat: add google place id to places table

ai-n0808 committed:
fix: typescript error and data of newPlace

brianaqp commented on prisma/schema.prisma:
```diff
@@ -34,11 +34,17 @@ model Experiences {
   longitude    Decimal @db.Decimal(9, 6)
+  google_place_id String? @unique @db.VarChar(255)
+  google_map_uri String? @db.Text
+  website_uri String? @db.Text
```
Good idea to include some additional params. We may used it.

brianaqp commented on src/router/tourists/index.ts:
```diff
@@ -1,18 +1,66 @@
+      const regularOpeningHours = JSON.stringify(
+        googlePlacesResult.regularOpeningHours
+      );
+      const photos = JSON.stringify(googlePlacesResult.photos);
```
This won't work. Photos its a table with the base64_encoded information, the actual image

brianaqp commented on prisma/schema.prisma:
```diff
@@ -34,11 +34,17 @@ model Experiences {
+  google_place_id String? @unique @db.VarChar(255)
+  google_map_uri String? @db.Text
+  website_uri String? @db.Text
+  regular_opening_hours String? @db.Text
```
places.regular_opening hours would be an string? I remember that is a whole object in maps

Its ok to try to added it, but its not ok to only stored as a JSON file.

Check how the structure is and how we can added into our schema

brianaqp commented on prisma/schema.prisma:
```diff
@@ -34,11 +34,17 @@ model Experiences {
+  website_uri String? @db.Text
+  regular_opening_hours String? @db.Text
+  formatted_address String? @db.VarChar(255)
+  photos String? @db.Text
```
Why photos? We already have Photos tables to store relationship between a places and their images

brianaqp commented on src/router/tourists/index.ts:
```diff
@@ -1,18 +1,66 @@
+      const longitude = googlePlacesResult.location.lng;
+      const googleMapsUri = googlePlacesResult.googleMapsUri;
+      const websiteUri = googlePlacesResult.websiteUri;
+      const regularOpeningHours = JSON.stringify(
```
we need to look inside the google response to find a convenient way of store this information, and not just json

brianaqp commented on src/router/tourists/index.ts:
```diff
@@ -1,18 +1,66 @@
+        where: { google_place_id: googlePlaceId },
+      });
+
+      if (existingPlace) {
```
Good validation

brianaqp reviewed (changes requested):
In terms of getting the data, its good

In terms of storing images, don't

You will need to research how with your Places Text Search information, get the images per place, and store it to fit into the table, as "base64"

ai-n0808 committed:
fix: modify to properly handle the storage of place images in the Images table

ai-n0808 commented:
As we discussed, I modified my code to handle the storage of place images. 
- Added a helper function, fetchBase64FromUrl, to fetch images from URLs and encode them as Base64.
- The updated code follows the requirement where for every place and every image reference (up to a maximum of 5 photos), you request the image and store it in Base64.

Also I added "onDelete: Cascade" to schema but I'm still not sure whether we need it or not.

brianaqp commented:
It looks fine, just let me go home and give it a better look

brianaqp commented on src/router/tourists/index.ts:
```diff
@@ -63,4 +76,11 @@ touristsRouter.post(
 
+// Utility function
+async function fetchBase64FromUrl(url: string): Promise<string> {
+  const response = await fetch(url);
```
This won't work. You are making a request. You need to research more about how this API works, make more testing to see whats inside this value, and look how you can retrieve the image.

brianaqp reviewed (approved):
@ai-n0808 
I will approved to start looking into this new changes you made, thanks to work in adding the new schema and making the google api for places. 

But we still need to work on that utility. Use this [resource](https://developers.google.com/maps/documentation/places/web-service/photos) for learn how to get the images with the **Places Photos Api**
